### PR TITLE
Add customizable Stop to provider

### PR DESCRIPTION
**Pull request by paultyng**

### CODE REVIEW

File: helper/schema/provider.go [hunk 1]
```diff
@@ -60,11 +60,20 @@ type Provider struct {
 	// provider doesn't need to be configured, this can be omitted.
 	ConfigureContextFunc ConfigureContextFunc
 
+	// StopFunc is called from the gRPC server Stop call. This does not
```
Comment (appilon) [inline helper/schema/provider.go:63]: Only nitpick is the wording here, I find the phrase `Terraform graph` a bit ambiguous, I would suggest something like
```
StopFunc is called from the gRPC server Stop call. This is an opportunity to cleanup things outside of Terraform's resource lifecycle and management (ie. session files, etc).
```
I'm no wordsmith though, what is here is already pretty good.

Comment (paultyng) [reply helper/schema/provider.go:63]: Want to write this up as a suggestion and I can merge it?

**Comment by paultyng** [comment]
Related to #63

**Comment by appilon** [comment]
Unfortunately after discussion we want to offer a proper shutdown hook but it can't be done through gRPC stop message. Closing until a shutdown hook makes it's way into the protocol or through `go-plugin`

**Pull request closed without merging**